nixos/modules/services/networking/tayga.nix

```diff
@@ -0,0 +1,162 @@
+{ config, lib, pkgs, ... }:
+
+with lib;
+let
+  cfg = config.services.networking.tayga;
+
+  # Converts an address set to a string
+  strAddr = addr: "${addr.address}/${toString addr.prefixLength}";
+
+  configFile = pkgs.writeText "tayga.conf" ''
+    tun-device ${cfg.tunDevice}
+
+    ipv4-addr ${cfg.ipv4.address}
+    ${optionalString (cfg.ipv6.address != null) "ipv6-addr ${cfg.ipv6.address}"}
+
+    prefix ${strAddr cfg.ipv6.pool}
+    dynamic-pool ${strAddr cfg.ipv4.pool}
+    data-dir ${cfg.dataDir}
+  '';
+
+  makeRouteCommands = oper: ''
+    ip addr ${oper} ${cfg.ipv4.router.address} dev ${cfg.tunDevice}
+    ip route ${oper} ${strAddr cfg.ipv4.pool} dev ${cfg.tunDevice}
+    ip addr ${oper} ${cfg.ipv6.router.address} dev ${cfg.tunDevice}
+    ip route ${oper} ${strAddr cfg.ipv6.pool} dev ${cfg.tunDevice}
+  '';
```

Comment:
Could we use `networking` options to implement this instead of ad-hoc iproute2 command?

Reply:
I already talked to hax404 about this and proposed to use networkd units instead. I'm opposed to relying on the highly undermaintained scripted networking parts of nixos.